xbmc/FileItem.cpp

```diff
@@ -1579,6 +1579,7 @@ void CFileItem::UpdateInfo(const CFileItem &item, bool replaceLabels /*=true*/)
       m_videoInfoTag = new CVideoInfoTag;
     }
 
+    m_strDynPath = item.m_strDynPath;
```

Comment:
submit this bug fix separately please (can be bundled in the same as change requested below).

Reply:
Also need to take care that everywhere that calls `CFileItem::UpdateInfo` it is OK for the strDynPath to possibly change, and that it has been correctly set in the item parameter.